cmd/system-probe/main.go

```diff
@@ -85,6 +87,14 @@ func runAgent(exit <-chan struct{}) {
 		gracefulExit()
 	}
 
+	if cfg.ProfilingEnabled {
+		if err := enableProfiling(cfg); err != nil {
+			log.Criticalf("failed to enable profiling: %s", err)
+			cleanupAndExit(1)
+		}
```

Comment:
should this really be a fatal error ?

Reply:
If you expected profiling to be running, and it wasn't, would you expect system-probe to keep going?

Reply:
When would this fail ?  There are a couple of features that can fail to get initialized and won't crash the system-probe (the various bpf modules, and conntrack.)  Profiling feels like a sufficiently "non-core" feature that it should not crash the system-probe.